### ISSUE TITLE
Use templates and components

### ISSUE THREAD

**Issue by martendegroot** [cookiemonster/divd.community#18]
Across the project, templates are re-using snippets of code (like the navigation menu). These could be organized into components or generic templates for improved maintenance and readability.

**Comment by martendegroot** [comment]
Heyy, just wondering. I've taken a closer look at the repo and it looks like the majority still isn't directly accessible through direct links, correct? For example the `/chapters` page.

Initially I wanted to stick to your current direction as much as possible and make components in vanilla JS and web components. That won't allow to move re-usable components (like the navigation bar for example) to a separated file though. Or at least not in plain HTML. There's still two workarounds that both aren't really beautiful imo. First using iframes to load the component from another HTML file, or dynamically creating the HTML [like so, in an example MDN provided](https://github.com/mdn/web-components-examples/blob/main/popup-info-box-web-component/main.js).

One other alternative to more easily scale the website would be to use a framework. Some tastes in that would be: React, or rather NextJS as mainstream option, SvelteKit for minimal footprint or Qwik as fast new kid on the block. All of these will add a layer of complexity though. But it will probably pay off on the long run. In the same street there's an option to piece together the website with a JS task runner like Grunt and keep everything quite vanilla beyond that. But I'm not sure that will make things much easier and readable.

So in short, the options:

1. Keep things as they are. Ignore the issue of having to update re-used components in multiple places.
2. Try to solve this with vanilla JS and HTML (and possibly smth like Grunt)
3. Use a framework

What do you think?
